packages/dd-trace/src/plugins/instrument.js

```diff
@@ -0,0 +1,122 @@
+'use strict'
+
+const dc = require('diagnostics_channel')
+const path = require('path')
+const semver = require('semver')
+const iitm = require('../iitm')
+const ritm = require('../ritm')
+const parse = require('module-details-from-path')
+const requirePackageJson = require('../require-package-json')
+const { AsyncResource } = require('async_hooks')
+
+const pathSepExpr = new RegExp(`\\${path.sep}`, 'g')
+const channelMap = {}
+exports.channel = function channel (name) {
+  const maybe = channelMap[name]
+  if (maybe) return maybe
+  const ch = dc.channel(name)
+  channelMap[name] = ch
+  return ch
+}
+
+exports.addHook = function addHook ({ name, versions, file }, hook) {
+  file = filename(name, file)
+  const loaderHook = (moduleExports, moduleName, moduleBaseDir) => {
+    moduleName = moduleName.replace(pathSepExpr, '/')
+    const moduleVersion = getVersion(moduleBaseDir)
+    if (moduleName !== file || !matchVersion(moduleVersion, versions)) {
+      return moduleExports
+    }
+    return hook(moduleExports)
+  }
+  ritm([name], loaderHook)
+  cjsPostLoad({ name, versions, file }, hook)
+  iitm([name], loaderHook)
+}
+
+function matchVersion (version, ranges) {
+  return !version || (ranges && ranges.some(range => semver.satisfies(semver.coerce(version), range)))
+}
+
+function getVersion (moduleBaseDir) {
+  if (moduleBaseDir) {
+    return requirePackageJson(moduleBaseDir, module).version
+  }
+}
+
+function filename (name, file) {
+  return [name, file].filter(val => val).join('/')
+}
+
+// TODO this is basically Loader#_getModules + running the hook. DRY up.
+// TODO delete all this as a semver major
```

Comment:
Can you add a line explaining why this is semver major?

Reply:
I just went ahead and deleted this line since it doesn't make enough sense on its own.